File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/responses/PositionResponse.java
package com.mindthecode.CompanyDirectory.models.responses;

import com.mindthecode.CompanyDirectory.models.entities.Unit;

public class PositionResponse {
    private long id;
    private String name;
    private Unit unit;

    public PositionResponse(long id, String name, Unit unit) {
        this.id = id;
        this.name = name;
        this.unit = unit;
    }

    public PositionResponse() {
    }

    public PositionResponse(long id, String name) {
        this.id = id;
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Unit getUnit() {
        return unit;
    }

    public void setUnit(Unit unit) {
        this.unit = unit;
    }

    @Override
    public String toString() {
        return "PositionResponse{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", unit=" + unit +
                '}';
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/position_response.dart
import 'package:company_directory_app/models/entities/unit.dart';

class PositionResponse {
  final int id;
  final String name;
  final Unit unit;

  PositionResponse({this.id, this.name, this.unit});

  factory PositionResponse.fromJson(Map<String, dynamic> json) {
    return PositionResponse(id: json['id'], name: json['name'], unit: Unit.fromJson(json['unit']));
  }
}
